src/IdentityServer/Configuration/DependencyInjection/Options/DynamicProviderOptions.cs

```diff
@@ -31,7 +31,24 @@ public class DynamicProviderOptions
     /// <summary>
     /// Scheme for signout. Defaults to the constant IdentityServerConstants.DefaultCookieAuthenticationScheme.
     /// </summary>
-    public string SignOutScheme { get; set; } = IdentityServerConstants.DefaultCookieAuthenticationScheme;
+    public string SignOutScheme 
+    { 
+        get 
+        {
+            return _signOutScheme ?? IdentityServerConstants.DefaultCookieAuthenticationScheme;
+        } 
+        set 
+        {
+            _signOutScheme = value;
+        }
+    }
+    
+    private string? _signOutScheme;
+
+    /// <summary>
+    /// Gets a value indicating if the SignOutScheme was set explicitly, either by application logic or by options binding.
+    /// </summary>
+    public bool SignOutSchemeSetExplicitly { get => _signOutScheme != null; }
```

Comment:
The options class now understands if the sign out scheme has actually been set. We use that in our configuration so that we don't override a user who is trying to set the scheme explicitly.